src/main/java/net/mcreator/ui/modgui/VillagerProfessionGUI.java

```diff
@@ -0,0 +1,210 @@
+
+		JButton importProfessionTexture = new JButton(UIRES.get("18px.add"));
+		importProfessionTexture.setToolTipText(L10N.t("elementgui.villager_profession.import_profession_texture"));
+		importProfessionTexture.setOpaque(false);
+		importProfessionTexture.addActionListener(e -> {
+			TextureImportDialogs.importMultipleTextures(mcreator, TextureType.OTHER);
+			professionTextureFile.removeAllItems();
+			professionTextureFile.addItem("");
+			mcreator.getFolderManager().getTexturesList(TextureType.OTHER)
+					.forEach(el -> professionTextureFile.addItem(el.getName()));
+		});
+
+		subpanel.add(HelpUtils.wrapWithHelpButton(this.withEntry("villagerprofession/profession_texture"),
+				L10N.label("elementgui.villager_profession.profession_texture")));
+		subpanel.add(PanelUtils.centerAndEastElement(professionTextureFile, importProfessionTexture));
+
+		page1group.addValidationElement(name);
+		page1group.addValidationElement(pointOfInterest);
+		page1group.addValidationElement(actionSound.getVTextField());
+		page1group.addValidationElement(professionTextureFile);
+
+		name.setValidator(new TextFieldValidator(name, L10N.t("elementgui.villager_profession.profession_needs_name")));
+		name.enableRealtimeValidation();
+		pointOfInterest.setValidator(new MCItemHolderValidator(pointOfInterest));
+		actionSound.getVTextField().setValidator(new TextFieldValidator(actionSound.getVTextField(),
+				L10N.t("elementgui.common.error_sound_empty_null")));
+		professionTextureFile.setValidator(() -> {
+			if (professionTextureFile.getSelectedItem() == null || professionTextureFile.getSelectedItem().equals(""))
+				return new Validator.ValidationResult(Validator.ValidationResultType.ERROR,
+						L10N.t("elementgui.villager_profession.profession_needs_texture"));
+			return Validator.ValidationResult.PASSED;
+		});
+
+		panel.add(PanelUtils.totalCenterInPanel(subpanel));
+		panel.setOpaque(false);
+
+		clo.setPreferredSize(new Dimension(320, 320));
+
+		JPanel clop = new JPanel();
+		clop.setOpaque(false);
+		clop.add(clo);
+
+		JPanel mainPanel = new JPanel(new BorderLayout(35, 35));
+		mainPanel.add("Center", panel);
+		mainPanel.add("South", clop);
+		mainPanel.setOpaque(false);
+
+		addPage(L10N.t("elementgui.common.page_properties"), mainPanel);
+
+		if (!isEditingMode()) {
+			name.setText(modElement.getName().toUpperCase(Locale.ROOT));
+		}
+	}
+
+	private void updateProfessionTexturePreview() {
+		if (professionTextureFile.getSelectedItem() == null)
+			return;
+		File professionTexture = mcreator.getFolderManager()
+				.getTextureFile(professionTextureFile.getSelectedItem(), TextureType.OTHER);
+		Image image = ImageUtils.resize(new ImageIcon(professionTexture.getAbsolutePath()).getImage(), 320, 320);
+		BufferedImage bg = new BufferedImage(image.getWidth(null), image.getHeight(null), BufferedImage.TYPE_INT_ARGB);
+		Graphics2D graphics2D = bg.createGraphics();
+		graphics2D.drawImage(image, 0, 0, image.getWidth(null), image.getHeight(null), null);
```

Comment:
Still do not understand why you need to re-draw the same image here. Please elaborate

Reply:
I'm not "re-drawing" anything

Reply:
`graphics2D.drawImage(image, 0, 0, image.getWidth(null), image.getHeight(null), null);`

What is this then?

Reply:
Well it draws the image, but it's only called once

Reply:
It re-draws image in bg, why is this needed?

Reply:
I tried removing lines 165 to 168 and replace old line 169 with this
`clo.setIcon(new ImageIcon(ImageUtils.resize(new ImageIcon(professionTexture.getAbsolutePath()).getImage(), 320, 320)));` but now, the texture doesn't appear..
<img width="904" alt="Capture d’écran 2022-05-19 à 13 59 39" src="https://user-images.githubusercontent.com/49318596/169288541-9f414efc-ebd2-4d96-bb18-f31fcac4c76d.png">
.

Reply:
Then the root cause of this needs to be fixed. Do you get any error logs?

Reply:
No error, the code that you want me to change works fine so I don't get why you want me to change it

Reply:
What I did works. If you don't understand how it works, then why did you use that code in a method used by the Armor mod element?

Reply:
The excuse of old bad code is not reasoning for new bad code. We will figure this out and make this proper on both armor and here

Reply:
Can we do that in another PR and let the code that I did here? I mean for the moment, it works, maybe we can make a different code for both armor and here in another PR?